src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1232,6 +1278,44 @@ class spell_warr_vigilance_trigger : public SpellScriptLoader
         }
 };
 
+// 1680 - Whirlwind 
+// Updated 7.1.5
+class spell_warr_whirlwind : public SpellScriptLoader
+{
+public:
+    spell_warr_whirlwind() : SpellScriptLoader("spell_warr_whirlwind") { }
+
+    class spell_warr_whirlwind_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warr_whirlwind_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_TRAUMA_EFFECT
+            });
+        }
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            if (GetCaster()->HasAura(SPELL_WARRIOR_TRAUMA))
+                //GetCaster()->CastSpell(GetHitUnit(), SPELL_WARRIOR_TRAUMA_EFFECT, true);
+                GetCaster()->CastSpell(GetHitUnit(), SPELL_WARRIOR_TRAUMA, true);
```

Comment:
typo here?
You validate SPELL_WARRIOR_TRAUMA_EFFECT, and cast SPELL_WARRIOR_TRAUMA again? No make sense to me